src/file-events.c

```diff
@@ -196,17 +198,20 @@ static __always_inline void exit_mkdir(struct pt_regs *ctx, tail_call_slot_t tai
     fm->u.action.buffer_len = sizeof(file_message_t);
     fm->u.action.u.create.source_link = LINK_NONE;
 
-    cached_path->next_dentry = event.target_dentry;
+    init_filtered_cached_path(cached_path, event.target_dentry, event.target_vfsmount);
 
     ResolveName:
     ret = write_path(ctx, cached_path, &cursor, tail_call);
-    cached_path->next_dentry = NULL;
     if (ret < 0) goto EmitWarning;
+    if (cached_path->filter_state <= 0) goto NoEvent; // Didn't match watched path filter
 
-    write_null_char(cursor.buffer, cursor.offset);
+    bpf_probe_read_str(fm->u.action.filter_tag, MAX_TAG, cached_path->filter_tag);
```

Comment:
should we emit a warning if this fails?

Reply:
I'm not sure how it could fail, but I expect it would be obvious from the resulting contents of fm's filter_tag.  Also it will go away if we replace the tag with an integer.